packages/inter-protocol/src/auction/offerBook.js

```diff
@@ -93,30 +144,54 @@ export const prepareScaledBidBook = baggage =>
         const { records } = this.state;
         return [...records.entries(M.gte(toBidScalingComparator(bidScaling)))];
       },
+      publishOffer(record) {
+        const key = toScaledRateOfferKey(record.bidScaling, record.seqNum);
+
+        return E(getBidDataRecorder(key)).write(
+          harden({
+            bidScaling: record.bidScaling,
+            wanted: record.wanted,
+            exitAfterBuy: record.exitAfterBuy,
+            timestamp: record.timestamp,
+            balance: record.seat.getCurrentAllocation().Bid,
+            sequence: record.seqNum,
+          }),
+        );
+      },
+      publishOffers() {
+        const { records } = this.state;
+
+        for (const r of records.values()) {
+          this.self.publishOffer(r);
+        }
```

Comment:
Before we land this, let's make sure we don't do O(n) work like this.

Reply:
Also, I don't think this handles the case when bids go away.

Until we have an explicit delete method in vstorage (#7405) I think `setValue('')` has essentially the same effect.